src/tokenizer/plugins/comments.ts

```diff
@@ -0,0 +1,50 @@
+import type MarkdownIt from 'markdown-it/lib';
+import type StateBlock from 'markdown-it/lib/rules_block/state_block';
+import type StateInline from 'markdown-it/lib/rules_inline/state_inline';
+
+const OPEN = '<!--';
```

Comment:
I'm totally peaceful with this approach, but I wonder if we should use the same regex that markdown-it uses internally: https://github.com/markdown-it/markdown-it/blob/df4607f1d4d4be7fdc32e71c04109aea8cc373fa/lib/common/html_re.js#L18

```js
var comment     = '<!---->|<!--(?:-?[^>-])(?:-?[^-])*-->';
```

Reply:
I'd prefer to avoid using a regex here